logger.js

```diff
@@ -121,7 +121,7 @@ function wrapReqSerializer (serializer) {
 function asReqValue (req) {
   var connection = req.connection
   const _req = Object.create(pinoReqProto)
-  _req.id = req.id
+  _req.id = Function.prototype.isPrototypeOf(req.id) ? req.id() : req.id
```

Comment:
why not `typeof req.id === 'function'`? Which one is faster?

Reply:
unless there's a speed benefit.. definitely go with `typeof` here...

Reply:
/shrug, I prefer using the actual API. I haven't tested for any difference in speed.